pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -72,8 +73,11 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	if err := c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner); err != nil {
-		return err
+	r.watchCheCluster = func() error {
+		if err := c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner); err != nil {
```

Comment:
you could replace the whole func body with 

```suggestion
		return c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner)
```

Reply:
Done https://github.com/nurali-techie/toolchain-operator/pull/1/commits/c78380be8bd9b9d4e5f6dab5a8a55bf2ef288b33